### PR TITLE
0037-Expand-Mobile-putfile-RPC

### PR DESCRIPTION
**Pull request by Jack-Byrne**
[Link to proposal](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0037-Expand-Mobile-putfile-RPC.md)

### CODE REVIEW

**Comment by theresalech** [comment]
Resolves #40

File: MOBILE_API.xml [hunk 1]
```diff
@@ -135,6 +135,9 @@
         <element name="READ_ONLY" since="4.5">
             <description>The value being set is read only</description>
         </element>
+        <element name="CORRUPTED_DATA" since="4.6">
```
Comment (theresalech) [inline MOBILE_API.xml]: I think this should be `since="5.0"`

File: MOBILE_API.xml [hunk 2]
```diff
@@ -5110,6 +5113,9 @@
                 If offset is set to 0, then length is the total length of the file to be downloaded
             </description>
         </param>
+        <param name="crc" type="Integer" minvalue="0" maxvalue="4294967295" mandatory="false" since="4.6">
```
Comment (theresalech) [inline MOBILE_API.xml]: I think this should be `since="5.0"`

File: MOBILE_API.xml [hunk 3]
```diff
@@ -5129,6 +5135,7 @@
             <element name="GENERIC_ERROR"/>
             <element name="REJECTED"/>
             <element name="UNSUPPORTED_REQUEST"/>
+            <element name=" CORRUPTED_DATA" since="4.6"/>
```
Comment (theresalech) [inline MOBILE_API.xml]: I think this should be `since="5.0"`

**Review by theresalech — CHANGES_REQUESTED** [review]
@JackLivio it looks like there are a few version reference changes required.

**Comment by Jack-Byrne** [comment]
@theresalech Updated

**Review by theresalech — APPROVED** [review]

**Review by joeygrover — APPROVED** [review]

**Pull request merged**